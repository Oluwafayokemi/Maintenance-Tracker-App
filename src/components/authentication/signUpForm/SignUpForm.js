import React from 'react';
import history from '../../../util/history';

export default class SignUpForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      firstName: '',
      lastName: '',
      department: '',
      email: '',
      password: '',
      passwordCheck: '',
    };
  }

  handleChange = (event) => {
    this.setState({ [event.target.name]: event.target.value });
  }

  handleSubmit = (event) => {
    const { signUp } = this.props;
    event.preventDefault();
    signUp(this.state)
    .then((response) => {
      if (response.user.isAdmin){
        history.push('/admin')
      }
      else if (!response.user.isAdmin) {
        history.push('/user')
      }
    })
    .catch((err) => {
      err
    })
  }
  
  render() {
    return (
      <React.Fragment>
        <div className="col-6 allbody three">
          <h1>Create a new account</h1>
          <p>Its free and always will be.</p>
          <form className="form menu" id="signup-form" onSubmit={this.handleSubmit}>
            <div className="container">
              <label htmlFor="firstName">
                <b>First Name</b>
              </label>
              <input
                type="text"
                id="firstName"
                placeholder="Enter First Name"
                name="firstName"
                value={this.state.firstName}
                onChange={this.handleChange}
                required
              />

              <label htmlFor="lastName">
                <b>Last Name</b>
              </label>
              <input
                type="text"
                id="lastName"
                placeholder="Enter Last Name"
                name="lastName" value={this.state.lastName}
                onChange={this.handleChange}
                required
              />

              <label htmlFor="department">
                <b>Department</b>
              </label>
              <input
                type="text"
                id="department"
                placeholder="Department"
                name="department"
                value={this.state.department}
                onChange={this.handleChange}
                required
              />

              <label htmlFor="email">
                <b>Email</b>
              </label>
              <input
                type="email"
                id="email"
                placeholder="Enter email"
                name="email"
                value={this.state.email}
                onChange={this.handleChange}
                required
              />

              <label htmlFor="password">
                <b>Password</b>
              </label>
              <input
                type="password"
                id="password" placeholder="Enter Password"
                name="password"
                required
                minLength="5"
                maxLength="20"
                value={this.state.password}
                onChange={this.handleChange}
              />

              <label htmlFor="passwordCheck">
                <b>Confirm Password</b>
              </label>
              <input
                type="password"
                id="passwordCheck"
                placeholder="Confirm Password"
                name="passwordCheck"
                required
                minLength="5"
                maxLength="20"
                value={this.state.passwordCheck}
                onChange={this.handleChange}
              />

              <button type="submit" value="submit">
                Register
              </button>
            </div>
          </form>
        </div>
      </React.Fragment>
    );
  }
}
