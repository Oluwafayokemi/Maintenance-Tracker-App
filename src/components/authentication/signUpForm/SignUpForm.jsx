import React from 'react';

const SignUpForm = ({ handleChange, handleSubmit, value }) => (
  <React.Fragment>
    <div className="col-6 allbody three">
      <h1>Create a new account</h1>
      <p>Its free and always will be.</p>
      <form className="form menu" id="signup-form" onSubmit={handleSubmit}>
        <div className="container">
          <label htmlFor="firstName">
            <b>First Name</b>
          </label>
          <input
            type="text"
            id="firstName"
            placeholder="Enter First Name"
            name="firstName"
            value={value.firstName}
            onChange={handleChange}
            required
          />

          <label htmlFor="lastName">
            <b>Last Name</b>
          </label>
          <input
            type="text"
            id="lastName"
            placeholder="Enter Last Name"
            name="lastName"
            value={value.lastName}
            onChange={handleChange}
            required
          />

          <label htmlFor="department">
            <b>Department</b>
          </label>
          <input
            type="text"
            id="department"
            placeholder="Department"
            name="department"
            value={value.department}
            onChange={handleChange}
            required
          />

          <label htmlFor="email">
            <b>Email</b>
          </label>
          <input
            type="email"
            id="email"
            placeholder="Enter email"
            name="email"
            value={value.email}
            onChange={handleChange}
            required
          />

          <label htmlFor="password">
            <b>Password</b>
          </label>
          <input
            type="password"
            id="password"
            placeholder="Enter Password"
            name="password"
            required
            minLength="5"
            maxLength="20"
            value={value.password}
            onChange={handleChange}
          />

          <label htmlFor="passwordCheck">
            <b>Confirm Password</b>
          </label>
          <input
            type="password"
            id="passwordCheck"
            placeholder="Confirm Password"
            name="passwordCheck"
            required
            minLength="5"
            maxLength="20"
            value={value.passwordCheck}
            onChange={handleChange}
          />

          <button type="submit" value="submit">
            Register
          </button>
        </div>
      </form>
    </div>
  </React.Fragment>
);

export default SignUpForm;
