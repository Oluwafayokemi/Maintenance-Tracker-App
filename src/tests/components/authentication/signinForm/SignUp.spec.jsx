import React from 'react';
import { shallow } from 'enzyme';
import SignUpForm from '../../../../components/authentication/signUpForm/SignUpForm';

describe('Tests SignUpForm component', () => {
  const props = {
    handleChange: jest.fn(),
    handleSubmit: jest.fn(),
    value: {
      firstName: 'firsname',
      lastName: 'lastName',
      email: 'email',
      department: 'department',
      password: 'password',
      passwordCheck: 'passWordCheck',
    },
  };
  it('should render without errors', (done) => {
    const wrapper = shallow(<SignUpForm {...props} />);
    expect(wrapper).toMatchSnapshot();
    done();
  });
});
