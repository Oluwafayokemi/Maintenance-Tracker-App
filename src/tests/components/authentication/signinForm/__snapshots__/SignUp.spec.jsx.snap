// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests SignUpForm component should render without errors 1`] = `
<Fragment>
  <div
    className="col-6 allbody three"
  >
    <h1>
      Create a new account
    </h1>
    <p>
      Its free and always will be.
    </p>
    <form
      className="form menu"
      id="signup-form"
      onSubmit={[MockFunction]}
    >
      <div
        className="container"
      >
        <label
          htmlFor="firstName"
        >
          <b>
            First Name
          </b>
        </label>
        <input
          id="firstName"
          name="firstName"
          onChange={[MockFunction]}
          placeholder="Enter First Name"
          required={true}
          type="text"
          value="firsname"
        />
        <label
          htmlFor="lastName"
        >
          <b>
            Last Name
          </b>
        </label>
        <input
          id="lastName"
          name="lastName"
          onChange={[MockFunction]}
          placeholder="Enter Last Name"
          required={true}
          type="text"
          value="lastName"
        />
        <label
          htmlFor="department"
        >
          <b>
            Department
          </b>
        </label>
        <input
          id="department"
          name="department"
          onChange={[MockFunction]}
          placeholder="Department"
          required={true}
          type="text"
          value="department"
        />
        <label
          htmlFor="email"
        >
          <b>
            Email
          </b>
        </label>
        <input
          id="email"
          name="email"
          onChange={[MockFunction]}
          placeholder="Enter email"
          required={true}
          type="email"
          value="email"
        />
        <label
          htmlFor="password"
        >
          <b>
            Password
          </b>
        </label>
        <input
          id="password"
          maxLength="20"
          minLength="5"
          name="password"
          onChange={[MockFunction]}
          placeholder="Enter Password"
          required={true}
          type="password"
          value="password"
        />
        <label
          htmlFor="passwordCheck"
        >
          <b>
            Confirm Password
          </b>
        </label>
        <input
          id="passwordCheck"
          maxLength="20"
          minLength="5"
          name="passwordCheck"
          onChange={[MockFunction]}
          placeholder="Confirm Password"
          required={true}
          type="password"
          value="passWordCheck"
        />
        <button
          id="signUp-btn"
          type="submit"
          value="submit"
        >
          Register
        </button>
      </div>
    </form>
  </div>
</Fragment>
`;
